packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx

```diff
@@ -78,24 +78,27 @@ export const start = () => {
     }
 
     const error = window.__ERROR__
+    const id = uuid()
 
     return Promise.resolve()
         .then(() => new Promise((resolve) => loadableReady(resolve)))
         .then(() => {
             ReactDOM.hydrate(
                 <ExpressContext.Provider value={{}}>
-                    <Router>
-                        <DeviceContext.Provider value={{type: window.__DEVICE_TYPE__}}>
-                            <AppConfig locals={locals}>
-                                <Switch
-                                    error={error}
-                                    appState={window.__PRELOADED_STATE__}
-                                    routes={routes}
-                                    App={WrappedApp}
-                                />
-                            </AppConfig>
-                        </DeviceContext.Provider>
-                    </Router>
+                    <CorrelationIdProvider correlationId={id}>
```

Comment:
We previously talked about the implications of up coming changes with `withLegacyGetProps` and the location of the `route-component`. One of those details is being able to reset the correlation id on "virtual" page views. 

Since that logic probably doesn't make sense to be in the `route-component` we could alternatively determine virtual page views as long as we have the route component as a parent. E.g. we can use the withRoute or useRoute to check for route changes and update the correlation id when required.

This might mean changing the correlationProvider tho, so it can generate an id internally if one isn't provided.